packages/kotti-ui/source/kotti-field-select/KtFieldSingleSelectRemote.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+	<div>
+		<div ref="ktFieldRef">
+			<KtField
+				v-bind="{ field }"
+				class="kt-field-select kt-field-select-single-remote"
+				:getEmptyValue="() => null"
+				:helpTextSlot="$slots.helpText"
+			>
+				<SingleSelectInput
+					:label="selectedLabel"
+					:placeholder="placeholder"
+					:queryString="queryValue"
```

Comment:
this is used for searching on the backend, correct? if so, how about making a debounce configurable as a prop to the SingleSelectInput, so that queryInputChanged emit is debounced and the remote is not hit when typing a word?

Reply:
I'd say that's on the user-app, not on Kotti. The component should remain purely controlled, it emits when it needs to, if the user-app wants to ignore the events, it debounces.

Reply:
Agreeing with @carsoli , it is not necessarily true - although likely - that this component is used in an async context or that the search operation is costly.  Debouncing should  be handled by whoever is implementing this.

Reply:
yeah. I was assuming it is always true, and probably it will be. I agree it is a bell/whistle, but 99% of the times it is used, so why not make it dry and let the component handle it was and is my thinking. By the way, it does not mean a debounce is always used, default value can be that it is 0. It is not a blocker for me, just wanted to point out the user app is going to always implement a debounce from the outside now for a component that by it's definition seems to always need one.

Reply:
if all the usages on user-app would debounce, it's on user-app to implement either a utility (hook) that handles the emitted event, debounces, makes the BE call, passes back a flag that it's loading/fetching and then the negation when it's done, and the data that it receives (so the options) and in this case it would also disable the select while loading from accepting further query, if it has to.

All of this ^ is not to show off but to state that it's not going to be just the "debounce" behavior that's needed but rather a whole logical implementation, and this is more and more proof that it's fit to be done on the user app
A custom component instead of the utility can be implemented as well.

I still disagree with the debounce proposal, but don't mind if another argument can be raised that tilts my opinion in favor of this.
@mvangent

Reply:
Like I said, not a blocker, so please go ahead as you see fit, I trust you find the best solution @carsoli and I don't disagree with your argumentation. Debounce is just a good way of making sure the user has enough time to search for the term they are actually searching for imo. And since this component is called selectremote, and a need for delay seemed implied by definition I wanted to raise it as a point. I can see why you want to do it on the wrapping layer. Thanks for sharing your thinking on it.

Reply:
I can't deny that other libraries don't already make this mistake but in essence they defy the whole point of this being a controlled component fully when they fiddle with the logic internally, and the way I see it it's always going to be a source of bugs at one point.

I can take the responsibility on Yoda's side or one of you guys @HansAuger @FlorianWendelborn even, after the urgent change is made for internal order ofc